### ISSUE TITLE
Problems with LectinOracle regression training

### ISSUE THREAD

**Issue by Old-Shatterhand** [BojarLab/glycowork#13]
Hi,

when retraining LectinOracle in a regression setting using the `train_model(...)` function I'm getting an error stating a problem with appending to a list in line 137 of `glycowork/ml/model_training.py`:
```python
running_acc.append(y.cpu().detach().numpy(), pred.cpu().detach().numpy())
```
The problem is that `append` only takes one argument, not two as provided in that line.

A code example (with the `train_loader` and `val_loader` filled as in your [LectinOracle Notebook](https://github.com/BojarLab/LectinOracle/blob/main/LectinOracle_GitHub.ipynb)):
```python
model = LectinOracle(input_size_prot=1280, input_size_glyco=len(lib) + 1, 
        hidden_size=128, num_classes=1, data_min=data_min, data_max=data_max)
model.apply(init_weights)
model.cuda()

optimizer = optim.Adam(model.parameters(), lr=0.0005, weight_decay=0.0)
scheduler = optim.lr_scheduler.CosineAnnealingLR(optimizer, 80)
criterion = nn.MSELoss().cuda()

model = train_model(model=model, dataloaders={"train": train_loader, "val": val_loader}, 
        criterion=criterion, optimizer=optimizer, scheduler=scheduler, num_epochs=100, 
        patience=20, mode="regression")
```

Compared to the metrics-handling for the classification setting there is missing the function call to `compute_accuracy` as in line 133: 
```python
running_acc.append(accuracy_score(y.cpu().detach().numpy().astype(int), pred2))
```
But as accuracy is not a suitable metric in a regression training, it might be better to rework the metrics for regression training in general (in the following, only accuracy and matthews coefficient are reported, both classification metrics).

Best, Roman

**Comment by Bribak** [comment]
Hi Roman,

I think you might be referring to an old version of the function (or the package) because line 137 in the current build is a blank line in glycowork/ml/model_training.py (https://github.com/BojarLab/glycowork/blob/master/glycowork/ml/model_training.py#L137)

The line that you were referring to is presumably line 164 in the current build:

`running_acc.append(mean_squared_error(y.cpu().detach().numpy(), pred.cpu().detach().numpy()))`

Which does append only one thing & uses MSE as a metric (we still call the variable "running_acc" but the actual metric is different depending on the task)

Does that help?
Daniel

**Comment by Old-Shatterhand** [comment]
Hi Daniel,

yes, this helps a lot, I was still in the code of version 0.3. I didn't updated all version of glycowork. Sorry for that.

Best, Roman
